src/utils/media-devices-manager.js

```diff
@@ -191,6 +200,36 @@ export default class MediaDevicesManager {
     }
   }
 
+  /**
+   * Webkit issue on iOS>13: https://bugs.webkit.org/show_bug.cgi?id=213853
+   */
+  _iOSTrackMutedHack() {
+    const iOSVersion = parseInt(browser.version.split(".")[0]);
+    if (
+      ["iOS", "Mac OS"].includes(detectedOS) &&
+      ["safari", "ios"].includes(browser.name) &&
+      iOSVersion >= IOS_MUTED_HACK_MIN_VERSION
+    ) {
+      this.audioTrack.onmute = this._muteStateChanged;
+      this.audioTrack.onunmute = this._muteStateChanged;
+    }
+  }
+
+  _muteStateChanged() {
+    if (this._audioTrack && this._audioTrack.muted) {
+      if (!this.reconnectIfUnmuted) {
+        NAF.connection.adapter.enableMicrophone(false);
+      }
+      this.reconnectIfUnmuted = true;
+    } else {
+      if (this.reconnectIfUnmuted) {
+        this.reconnectIfUnmuted = false;
+        this.setMediaStreamToDefault();
+        NAF.connection.adapter.enableMicrophone(true);
+      }
```

Comment:
Instead of 
```
this.setMediaStreamToDefault();
NAF.connection.adapter.enableMicrophone(true);
```
you should call 
```
this.recreateAudioStream()
```
so it calls getUserMedia again with the previous deviceId and reexecute `_iOSTrackMutedHack()`
(`setMediaStreamToDefault()` is not calling again `_iOSTrackMutedHack()` here)
and be sure it calls `enableMicrophone(currentMicStatus)`.
If the user was muted, then answered a Skype call, you don't want to enable the mic after he finished his Skype call.

You probably need to merge #3543 first to expose recreateAudioStream on `this`.
You may want to really have and exact match (not the default "ideal" match) in recreateAudioStream too, like I did in my PR https://github.com/mozilla/hubs/pull/3543 see the comments in the code.

Reply:
That's what `setMediaStreamToDefault` does using the last used mic device, it ends up calling `fetchAudioTrack` which recreates the track and calls `_iOSTrackMutedHack`.

Reply:
I'm going to update to handle the case when the mic was muted before the mute state change so we don't unmute.

Reply:
You're absolutely right, it calls `fetchAudioTrack`, all good.